Add new fraud protection section to WCPay settings

allie500 opened the pull request:
Fixes Automattic/woocommerce-payments-server#2796 on the server repo.

#### Changes proposed in this Pull Request

<!--
Title: A descriptive, yet concise, title.
-->

Add new fraud protection section to WCPay settings

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->
***Note: This section is currently behind a feature flag. See PR #5645 for info on how to enable the fraud and risk tools feature flag.***

This PR adds the standard and high fraud protection settings section user interface to the WCPay settings screen. This new section will allow merchants to select between the standard ruleset and the high ruleset. The new section also features a modal to provide information about each ruleset.

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Check out this branch.
* All tests should pass.
* Login to your local client site and make sure your store's currency is set to USD.
* Visit the WCPay settings screen: `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments`
* Scroll to the bottom of the page. If you don't see the fraud protections section below the deposits section, then the feature flag is not enabled. See below for the command to run to enable it.

![No fraud protection settings section rendered](https://user-images.githubusercontent.com/60988591/219794624-3c11abee-4d24-4b4e-b48e-5b135f9679c8.png)

* Run the following command to enable the section to be rendered: `npm run wp option add wcpay_fraud_protection_settings_active 1`
* Refresh the settings page and scroll to the bottom. The fraud protection section should now be rendered below the deposits section.

![Fraud protection section rendered below deposits](https://user-images.githubusercontent.com/60988591/219795464-232c07bc-f735-4275-ab37-da5e8f208982.png)

* Evaluate the section: 
  * Check that there are no errors in the browser console. 
  * Check the text and layout for issues. 
  * Click the question mark icon beside each option to ensure that the correct modal displays. 
  * Ensure that modals close when clicking 'Got it' button, 'X' close button, and clicking outside of the modal.

* Navigate to the WooCommerce general settings page and change you currency to a currency other than USD. 
* Navigate back to the WCPay settings page and open each fraud protection modal. The rule referencing `$1,000.00` should now show a slightly modified text mentioning the equivalent of $1,000.00 in `$store_currency`.
 
* Check the section on mobile. Tunnel your local client site so you can reach it via the public internet.
* Visit your client site on your smartphone and login as the administrator.
* Navigate to the WCPay settings screen and scroll to the bottom of the page. 
* Evaluate the section on mobile: 
  * Check the text and layout for issues. 
  * Click the question mark icon beside each option to ensure that the correct modal displays. 
  * Ensure that modals close when clicking 'Got it' button and the 'X' close button.

Once you are done testing, if you would like to disable the feature flag, simply set the option to `0` or delete the option:
* `npm run wp option update wcpay_fraud_protection_settings_active 0`
* `npm run wp option delete wcpay_fraud_protection_settings_active`

***Note: The "Learn more about risk filtering" link also does not currently link anywhere as the documentation has yet to be published.***

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

allie500 committed:
Add fraud prevention settings card

allie500 committed:
Add fraud protection level hook

allie500 committed:
Add fraud protection section to settings

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Update icon used in modal notice

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Add initial fraud protection settings unit test

allie500 committed:
Add TipIcon to new icons dir

Remove TipIcon component from fraud prevention

Import TipIcon from new icons dir

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Add dollar amount rule components

Shows different line item depending on level and store currency

allie500 committed:
Fix failing unit test

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Consolidate notice components

eduardoumpierre commented on client/settings/fraud-protection/index.js:
```diff
@@ -0,0 +1,501 @@
+		);
+	}
+
+	return (
```
As both paths share the same initial component structure, it could be rewritten to something like this:

```js
const ExceedsDollarAmountRuleUSDStore = ( props ) => {
	const isHighProtectionLevel = 'high' === props.level;

	return (
		<li>
			{ __( 'An order exceeds', 'woocommerce-payments' ) }{ ' ' }
			<strong>{ __( '$1,000.00', 'woocommerce-payments' ) } </strong>

			{ ! isHighProtectionLevel && (
				<>
					{ __( 'or', 'woocommerce-payments' ) }{ ' ' }
					<strong>
						{ __( '10 items.', 'woocommerce-payments' ) }
					</strong>
				</>
			) }
		</li>
	);
};
```

allie500 replied:
I'm not sure this would work for the USD store component as the high level text has a period after the dollar amount: `'$1,000.00.'`

eduardoumpierre commented on client/settings/fraud-protection/index.js:
```diff
@@ -0,0 +1,501 @@
+	};
+
+	return (
+		<>
```
The Fragment `<>` can be removed as `<fieldset>` is the only direct child of it.

allie500 replied:
Thanks for catching that. Removed in [bc73d05](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/bc73d05e0a294a833d3644dc36932f949f31a250).

eduardoumpierre commented on client/settings/fraud-protection/index.js:
```diff
@@ -0,0 +1,501 @@
+	);
+};
+
+const ProtectionLevelModalNotice = ( props ) => {
```
Similarly to the previous one, we can merge the component structure and use ternary to determine the notice text.

```js
const ProtectionLevelModalNotice = ( { level } ) => {
	const isHighProtectionLevel = 'high' === level;

	return (
		<Notice
			className="component-notice--is-info"
			status="info"
			isDismissible={ false }
		>
			<div className="component-notice__content--flex">
				<TipIcon className="component-notice__icon" />
				<p>
					{ isHighProtectionLevel
						? __(
								'Offers the highest level of filtering for stores, but may catch some legitimate transactions',
								'woocommerce-payments'
						  )
						: __(
								"Provides a standard level of filtering that's suitable for most business.",
								'woocommerce-payments'
						  ) }
				</p>
			</div>
		</Notice>
	);
};
```

allie500 replied:
Thanks for pointing this out. Fixed in [0835f2c](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/0835f2cdda206552538b6a5acb66bdd083a8a83a).

eduardoumpierre commented on client/settings/fraud-protection/index.js:
```diff
@@ -0,0 +1,501 @@
+									onClick={ () => setHighModalOpen( true ) }
+								/>
+								{ isHighModalOpen && (
+									<Modal
```
It would be nice if we could extract those modals from the main component.

allie500 replied:
Agreed. I've added it to my to-do list.

allie500 replied:
Modals have been extracted and placed in their own file in [b769432](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/b769432e4a042907866aa10735c9e01e8f3462ef). Let me know what you think of the new implementation.

eduardoumpierre replied:
The changes are looking good! As a final nitpick, we can remove the fragment `<>` and return the short circuit directly:

```js
return isHighModalOpen && (<Modal ...>);
```

or add some logic before and then return the component:
```js
if ( ! isHighModalOpen ) {
  return null;
}

return <Modal ...>
```

The first solution is shorter but may backfire if the first operand `isHighModalOpen` doesn't return a boolean. You can easily get a `0` (it happens if you're doing something like `array.length && <>`) or an error because React can't render `undefined`. The idea is that it will return the value of the first operand if it's falsy, so `undefined && <Modal>` would return `undefined`.  This isn't the case with your code, but something to always keep in mind!

allie500 committed:
Migrate components to subdirectory

allie500 committed:
Refactor ProtectionLevelModalNotice component

allie500 committed:
Remove unneeded fragment

kalessil commented:
I was just passing by. Any reason why don't we use TypeScript for new files here?

Context for my question: paJDYF-6MV-p2

allie500 commented:
> I was just passing by. Any reason why don't we use TypeScript for new files here?
> 
> Context for my question: paJDYF-6MV-p2

Hi @kalessil. @eduardoumpierre and I will be converting these files to TypeScript during our next cooldown week. TypeScript is new for me so I still need to do a bit more learning before the cooldown week. Then Eduardo and I will pair up to knock it out.

allie500 committed:
Migrate help text into separate component

allie500 committed:
Update modal components

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

kalessil commented:
> Hi @kalessil. @eduardoumpierre and I will be converting these files to TypeScript during our next cooldown week. TypeScript is new for me so I still need to do a bit more learning before the cooldown week. Then Eduardo and I will pair up to knock it out.

Super! Thanks for keeping this on the radar.

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Restructure files and add unit tests

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Remove unneeded items.

allie500 committed:
Convert to snapshot tests

Commit snapshots

allie500 committed:
Add snapshot tests for fp-help-text component

Commit snapshots

allie500 committed:
Update text string

Add snapshot tests for component

Commit snapshots

allie500 committed:
Move ProtectionLevels to own file

Add snapshot test for ProtectionLevels

Commit snapshot

allie500 committed:
Add snapshot tests for fp-modal components
Commit snapshots

allie500 committed:
Convert FraudProtection test to snapshot
Commit snapshot

allie500 committed:
Move tests and snapshots to test subdirs

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Add todo comment to finish implementation

allie500 committed:
Put fraud protection settings behind feature flag
Add necessary unit tests

allie500 committed:
Add changelog file

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

jessepearson commented on client/settings/fraud-protection/components/exceeds-dollar-amount-rule/index.js:
```diff
@@ -0,0 +1,79 @@
+
+	return (
+		<li>
+			{ __( 'An order exceeds', 'woocommerce-payments' ) }{ ' ' }
```
Are we able to use `interpolateComponents` for all of these strings so that there aren't translations needed for each of the short strings?

allie500 replied:
Thanks for this suggestion! I was unaware of `interpolateComponents`. I think this would be a good use for it where we are simply adding bold or italics to the text. I'll look at getting this updates soon.

allie500 replied:
Updated in [8008338](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/8008338ba1612bf9b075251bd2f66598e992c5d9).

jessepearson commented on client/settings/fraud-protection/components/fp-modal/index.js:
```diff
@@ -0,0 +1,238 @@
+							<span className="component-modal__text--blocked">
+								{ __( 'blocked ', 'woocommerce-payments' ) }
+							</span>
+							{ __( 'if: ', 'woocommerce-payments' ) }
```
Another instance where we may be able to use `interpolateComponents` to generate one string. I am unsure if it's possible to pass a class name into the html, though.

allie500 replied:
I don't believe it is possible to pass a class name into the html, but I'll research further to find our for sure.

allie500 replied:
I researched many of the numerous uses of `interpolateComponents` in the codebase and you can definitely pass a class name within the components property and do all sorts of neat stuff in there. I fixed this in [385c894](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/385c894ff631f59b7012878f69e677a32337de59).

eduardoumpierre commented on client/settings/fraud-protection/style.scss:
```diff
@@ -0,0 +1,89 @@
+
+.component-modal__button {
+	&--confirm {
+		float: right;
```
The confirmation button is slightly off the modal's spacings due to the `float: right;`.

<img width="573" alt="image" src="https://user-images.githubusercontent.com/16882226/220922947-f4405128-fc31-4e01-b3b9-a3924142ca84.png">

I'd discourage the usage of `float` as it usually causes more damage than good 😄 

As a fix, we could try a different approach, such as using `margin-left: auto;` to align the button to the right:

```scss
.component-modal__button {
	&--confirm {
            display: block;
            margin-left: auto;
        }
}
```

The result would be:

<img width="573" alt="image" src="https://user-images.githubusercontent.com/16882226/220923074-6acc485b-e63d-485f-9313-388932a6b7f8.png">

allie500 replied:
Thanks for catching this and for the advice on `float`. I've updated this in [e9f1f32](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/e9f1f32f018306217e961086c1117ac87403aa66).

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Refactor to use interpolateComponents
Update snapshots

allie500 committed:
Refactor to use interpolateComponents
Update snapshots

allie500 committed:
Update modal confirm button style

tpaksu commented:
Updated the comment to include link to the server issue.

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Remove feature flag code
Moving to separate PR

tpaksu committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

tpaksu commented:
I've created my PR branch (#5638) on top of this branch, so I can continue building the advanced section on your design. FYI :)

tpaksu commented on client/settings/fraud-protection/components/protection-levels/index.js:
```diff
@@ -0,0 +1,118 @@
+		<fieldset>
+			<ul>
+				<li>
+					<label htmlFor="fraud-protection__standard-level">
```
Found an issue writing the advanced section, this label's `htmlFor` attribute needs to match the child input's `id` value to be able to get selected via clicking the text. It's not currently selectable via clicking the label.

allie500 replied:
Thanks for catching this! I have fixed it in [9303f59](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/9303f592a73c242240cfce8574f542c435756a61).

allie500 replied:
I need to work on this some more. I just noticed that now the modals don't open correctly any more as a result of this change. They require double clicking and the rendering looks off. I'll see what I can do.

allie500 replied:
Okay, fixed for real in [2d6aa3c](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/2d6aa3cdc05974a29d189a513404bb22e8b0f4bd).

tpaksu commented on client/settings/fraud-protection/components/protection-levels/index.js:
```diff
@@ -0,0 +1,118 @@
+					</label>
+				</li>
+				<li>
+					<label htmlFor="fraud-protection__high-level">
```
The above comment also applies to this.

allie500 replied:
Thanks for catching this! I have fixed it in [9303f59](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/9303f592a73c242240cfce8574f542c435756a61).

allie500 replied:
Okay, fixed for real in [2d6aa3c](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/2d6aa3cdc05974a29d189a513404bb22e8b0f4bd).

allie500 committed:
Set label and ID values to be the same
Update snapshots

allie500 committed:
Refactor for radio & modal compatibility
Update affected snapshots

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

eduardoumpierre commented on client/settings/fraud-protection/components/protection-levels/index.js:
```diff
@@ -0,0 +1,119 @@
+							onChange={ () => handleLevelChange( 'standard' ) }
+						/>
+						<label htmlFor="fraud-protection__standard-level">
+							<p className="fraud-protection-single-radio-wrapper__item">
```
What do you think about merging `<label>` and `<p>`? I think it would make sense after the latest structure changes as the `<p>` is the only direct child of `<label>`.

```js
<label className="fraud-protection-single-radio-wrapper__item" htmlFor="fraud-protection__standard-level">
```

allie500 replied:
Great suggestion. Thanks for catching this. Fixed in [e1a2864](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/e1a2864f2970322d6c65d2d562b92b93addfd395)

eduardoumpierre commented on client/settings/fraud-protection/style.scss:
```diff
@@ -0,0 +1,94 @@
+		display: flex;
+		align-items: center;
+		height: 20px;
+		margin-bottom: 8px;
```
What do you think about changing the margin to `4px`? When using the `20px` reference, the distance from the label is `4px` instead of `8px`.

<img width="519" alt="image" src="https://user-images.githubusercontent.com/16882226/222747286-d974f13e-4d2a-4a4b-b29f-1bb23dab57f4.png">

allie500 replied:
Thanks for catching all these little spacing issues I overlooked. Fixed in [e1a2864](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/e1a2864f2970322d6c65d2d562b92b93addfd395).

eduardoumpierre commented on client/settings/fraud-protection/style.scss:
```diff
@@ -0,0 +1,94 @@
+		height: 20px;
+		margin-bottom: 8px;
+		&__item {
+			margin-right: 5px;
```
I'd suggest resetting the top and bottom margins with `margin: 0 5px 0 0;` to remove the extra spacing on the label:

<img width="519" alt="image" src="https://user-images.githubusercontent.com/16882226/222747900-e59dcf38-f562-4dad-afe8-a12936ee0b0f.png">

allie500 replied:
Thanks for catching all these little spacing issues I overlooked. Fixed in [e1a2864](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/e1a2864f2970322d6c65d2d562b92b93addfd395).

eduardoumpierre commented on client/settings/fraud-protection/style.scss:
```diff
@@ -0,0 +1,94 @@
+		font-size: 12px;
+		color: #757575;
+		margin: 0 0 0 30px;
+		margin-bottom: 1em;
```
What do you think about removing the `margin-bottom: 1em` from the `help-text`? The Figma reference for the label distance is `16px`, and it works for the first one, but it creates extra spacing in the last item.

<table>
<tr>
<td>
<img width="365" alt="image" src="https://user-images.githubusercontent.com/16882226/222751178-08cc074b-40d7-4089-ae36-bd9233b59b70.png">

</td>
<td>
<img width="329" alt="image" src="https://user-images.githubusercontent.com/16882226/222751064-61901df3-b029-4595-a9aa-4eb53d5d0394.png">


</td>
</tr>
</table>

Instead of setting the margin in the `help-text`, it could be added to a more structural element like the `li`, which already have a CSS rule to remove the `margin-bottom` from the last `li` element:

```scss
.fraud-protection {
    &-single-radio-wrapper {
        ...
    }

    ul {
        margin-top: 0;

        > li {
            margin-bottom: 16px;
        }
    }

    input[type='radio'] {
...
```

The `margin-top: 0` on the `ul` aims to remove the extra spacing between the section title and the content.

<table>
<tr>
<td>
<img width="329" alt="image" src="https://user-images.githubusercontent.com/16882226/222750597-a810ee3b-6cc2-4f76-ba91-4019eb9052ac.png">

</td>
<td>
<img width="329" alt="image" src="https://user-images.githubusercontent.com/16882226/222750723-41f1c3b0-81f9-4d5c-b81a-2fda55164477.png">

</td>
</tr>
</table>

allie500 replied:
Thanks for the suggested fix here. It works great. Fixed in [e1a2864](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/e1a2864f2970322d6c65d2d562b92b93addfd395).

allie500 replied:
Fully fixed in [dbbf55f](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/dbbf55f4770c110bead37bb80632318dc99e3620). I forgot to remove the `margin-bottom: 1em`.

eduardoumpierre commented on client/settings/fraud-protection/index.js:
```diff
@@ -0,0 +1,26 @@
+	return (
+		<Card className="fraud-protection">
+			<CardBody>
+				<h4>{ __( 'Payment risk level', 'woocommerce-payments' ) }</h4>
```
The Figma layout is not too consistent with the section title, there are two variants:

- Payment risk level (WIM5w2Y10SCNBFdMLSVftj-fi-1300%3A197635)
- Set your payment risk level (WIM5w2Y10SCNBFdMLSVftj-fi-1590%3A234307)

Can we confirm with Brady which is the correct title?

allie500 replied:
@bradyv Can we get your input here for the Fraud Protection card title? 

Should it be: Payment risk level

Or should it be: Set your payment risk level

See comment above for additional info.

bradyv replied:
Sorry for the discrepancy, that's my fault for missing a few instances when updating the doc copy. It should be `Set your payment risk level` and I've updated the Figma file to match

bradyv replied:
@allie500 Sorry for the discrepancy, that's my fault for missing a few instances when updating the doc copy. It should be `Set your payment risk level` and I've updated the Figma file to match

allie500 replied:
@bradyv No worries! Thanks for clarifying. I'll get it updated.

allie500 replied:
I've updated the section's card title to `Set your payment risk level` in [d53bc92](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/d53bc922049dc4a4c856a8d0fbb55386d75cacdc).

However after I pushed that commit and reviewed Figma, I noticed an inconsistency in the setting's description to the left of the card, under the header `Fraud Protection`. @bradyv could you review the two items below and let me know which one is the correct one to use?

On this Figma layout (WIM5w2Y10SCNBFdMLSVftj-fi-1300%3A197505) it reads:
> Manage security and fraud prevention options to avoid chargebacks, unauthorized access to the store, and better store management experience.

On this Figma layout (WIM5w2Y10SCNBFdMLSVftj-fi-1300%3A197765) it reads: 
> Help avoid chargebacks by setting your security and fraud protection risk level.

elizaan36 replied:
hey @allie500 

This is the correct copy for the setting's description under `Fraud protection`:

> Help avoid chargebacks by setting your security and fraud protection risk level.

elizaan36 replied:
Just a quick note that the headers should all be sentence case (only the first word capitalized).

allie500 replied:
Hi @elizaan36, thanks for providing the correct settings description, and for catching the header capitalization issue. Both are fixed in [297bb3f](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/297bb3fe7a06fd2f0d2a461263b309d66343f856).

eduardoumpierre commented on client/settings/fraud-protection/components/fp-help-text/index.js:
```diff
@@ -0,0 +1,26 @@
+		<p className={ 'fraud-protection__text--help-text' }>
+			{ isHighProtectionLevel
+				? __(
+						'High protection: Offers the highest level of filtering for stores, ' +
```
Is it necessary to repeat the option name on each help text?

<img width="564" alt="image" src="https://user-images.githubusercontent.com/16882226/222752084-8b3f3d9d-5eac-4d9f-9367-b6e4be370c89.png">

([Figma reference](WIM5w2Y10SCNBFdMLSVftj-fi-1590%3A234307))

allie500 replied:
Thanks for catching this. When I made those strings, I was copying directly from Figma. Perhaps there was an update later on Figma and I just didn't catch it. I'll get those updated and committed.

allie500 replied:
Fixed in [e1a2864](https://github.com/Automattic/woocommerce-payments/pull/5450/commits/e1a2864f2970322d6c65d2d562b92b93addfd395).

eduardoumpierre reviewed (commented):
I followed the testing instructions and the feature flag worked as expected. The code is easy to follow and well-tested.

I've added minor suggestions to make your code look closer to the designs, and I think we can merge this PR after addressing those changes.

Congratulations on your work and thanks for working on this issue, Allie!

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Address latest round of feedback
Update affected snapshots

allie500 commented:
> I followed the testing instructions and the feature flag worked as expected. The code is easy to follow and well-tested.
> 
> I've added minor suggestions to make your code look closer to the designs, and I think we can merge this PR after addressing those changes.
> 
> Congratulations on your work and thanks for working on this issue, Allie!

Thanks for re-reviewing this. I've addressed all but one of the minor suggestions. I'm just waiting to hear back from Brady on the one remaining item.

allie500 committed:
Remove uneeded style

allie500 committed:
Update fraud protection card header
Update affected snapshot

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

tpaksu committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Update fraud protection description
Update unit tests

eduardoumpierre reviewed (approved):
I followed the testing instructions, and the currency reference and feature flag worked as expected.

The UI matches the Figma layout, the code is easy to follow and well-tested.

Thanks for working on this issue! LGTM :shipit:

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section

allie500 committed:
Merge branch 'develop' into add/2796-new-fraud-protection-section
